Alert from frontend errors in critical paths

andreslucena opened the pull request:
<!--
NOTE: We are in the middle of a big redesign of the frontend.
That could mean that your PR will not get through on the next few months.
Please see https://github.com/decidim/decidim/discussions/9512
-->

#### :tophat: What? Why?

During the voting process in Elections, there could be some exceptions, and these will not be shown to the participant. They are only visible in the console. 

#### :pushpin: Related Issues

- Supersedes #9356

#### Testing

1. Having a working Election. Things like #10900 and #10870 should ease up the process
2. Generate an exception on the try block. I've done with 
```diff
diff --git a/decidim-elections/app/packs/src/decidim/elections/voter/new-vote.js b/decidim-elections/app/packs/src/decidim/elections/voter/new-vote.js
index ca0ad26d60..2411519c89 100644
--- a/decidim-elections/app/packs/src/decidim/elections/voter/new-vote.js
+++ b/decidim-elections/app/packs/src/decidim/elections/voter/new-vote.js
@@ -21,6 +21,7 @@ $(async () => {
   // Get the vote component and bind it to all UI events
   const voteComponent = setupVoteComponent($voteWrapper);
   try {
+    throw new Error("Testing");
     await voteComponent.bindEvents({
       onBindEncryptButton(onEventTriggered) {
         $(".button.confirm").on("click", onEventTriggered);
```
3. See the error 

### :camera: Screenshots

![Screenshot of the new "Failed" screen](https://github.com/decidim/decidim/assets/717367/458abbcb-3c46-4a56-8c42-698c243046e4)


:hearts: Thank you!

alecslupu reviewed (dismissed):
When there is an error triggered by the js present in `new-vote.js`, the screen displayed the erros message. 
I could not test using the real vote algorithm due to some access code issues. I could reproduce the behaviour in the preview mode.

alecslupu commented:
@andreslucena This is approved, but i will leave it as it is for the moment. Your call when to merge it.

alecslupu commented:
@andreslucena #10804 has been merged, and now we have 3 conflicts on this PR.

andreslucena commented:
> https://github.com/decidim/decidim/pull/10804 has been merged, and now we have 3 conflicts on this PR.

@alecslupu I've merged with develop and solved the issues. I had to make a small change regarding the show/hide new API (see 3b393e0a7a57f4f4ab7e7b9e8aec91e92cb63753). 

I've updated it the Screenshot with the new error message. You can trust me on this one if you don't want to check it out again 😄  (as it's a pain to work with the bulletin-board, i.e. having to wait 1/2 hours just to try this small change). 

I've also added the `no-backport` label as `decidim-elections` isn't officially released and it's just too much effort to invest in something that no-one is using yet in production. We can apply this same principle to other PRs from `decidim-elections` too of course.

alecslupu reviewed (approved):
LGTM. 
Since (https://github.com/decidim/decidim/commit/3b393e0a7a57f4f4ab7e7b9e8aec91e92cb63753) is the only change to what i have previously approved, i am approving it in the current state :) .
